doc/lsst.pipe.base/creating-a-pipeline.rst

```diff
@@ -0,0 +1,448 @@
+individual contracts. Each list member may either be the ``contract``
+expression or a mapping of the expression and a message to raise if the
+contract is violated. If the contract is defined as a mapping, the expression
+is associated with a key named ``contract`` and the message is a simple string
+associated with a key named ``msg``.
+
+The expression section of ``contracts`` make reference to configuration
+parameters for one or more tasks. These expressions make use of the label
+assigned to a task in the ``tasks`` section to indicate which task a
+configuration parameter belongs to. The syntax is similar to that of a pex
+config file where the ``config`` variable is replaced with the task label
+associated with the task to configure. Take a look at a contract for the
+``DRP.yaml`` `Pipeline` for an example.
+
+.. code-block:: yaml
+
+    contracts:
+      - contract: "makeWarp.matchingKernelSize ==\
+                   assembleCoadd.matchingKernelSize"
+        msg: "The warping kernel size must be consistent between makeWarp and 
+              assembleCoadd tasks"
+
+It is important to note how ``contracts`` relate to ``parameters``. While a
+``parameter`` can be used to set two configuration variables to the same
+value at the time `Pipeline` definition is read, it does not offer any
+validation. It is possible for someone to change the configuration of one of
+the fields before a `Pipeline` is run. Because of this, ``contracts`` should
+always be written without regards to how ``parameters`` are used.
+
+-------
+Subsets
+-------
+
+`Pipelines` are the definition of a processing workflow from some input data
+products to some output data products. Frequently, however, there are sub
+units within a `Pipeline` that define a useful unit of the `Pipeline` to run
+on their own. This may be something like processing single frames only.
+
+You, as the author of the `Pipeline`, can define one or more of the
+processing units by creating a section in your `Pipeline` named ``subsets``.
+The value associated with the ``subsets`` key is a new mapping. The keys of
+this mapping will be the labels used to refer to an individual ``subset``.
+The values of this mapping can either be a yaml list of the tasks labels to
+be associated with this subset, or another yaml mapping. If it is the latter,
+the keys must be ``subset``, which is associated with the yaml list of task
+labels, and ``description``, which is associated with a descriptive message
+of what the subset is meant to do. Take a look at the following two examples
+which show the same ``subset`` defined in both styles.
+
+.. code-block:: yaml
+
+  subsets:
+    processCcd:
+      - isr
+      - characterizeImage
+      - calibrate
+
+.. code-block:: yaml
+
+  subsets:
+    processCcd:
+      subset:
+        - isr
+        - characterizeImage
+        - calibrate
+      description: A set of tasks to run when doing single frame processing
```

Comment:
The two formats are just so you can have a quick version (without description) and a long one with a description. What you wrote above is not valid. processCcd is the label for a subset, all task labels defined under it are associated with it. The reason there is the key `subset` under that is because it is like a dictionary, you need a key to associate with the list and differentiate it with the description key.

If you want a second subset, you put another label under subsets i.e.:
```
subsets:
  processCcd:
     ...
  coaddition:
    ...
```